### PR TITLE
Fix the cursor handling if the HW cursor is not working

### PR DESCRIPTION
**Pull request by pallavigopal**
Cursor is not positioned correctly when we fall back to the
3D composition fixed it by moving the cursor as a normal layer
and also handle like rest of input layers.

Now cursor can be assigned to overlay plane or 3D composition based
on the available input layers which will avoid unnecessary composition
due to cursor change every time.

Jira: AIA-334/IAHWC-64
Test: Now cursor and navigation bar is visible as expected in most of the apps

Signed-off-by: Pallavi G <pallavi.g@intel.com>

### CODE REVIEW

File: common/display/displayplanemanager.cpp [hunk 1]
```diff
@@ -195,6 +218,7 @@ std::tuple<bool, DisplayPlaneStateList> DisplayPlaneManager::ValidateLayers(
         // it.
         if (!FallbacktoGPU(j->get(), layer, commit_planes)) {
           composition.emplace_back(j->get(), layer, index);
+          const HwcRect<int> &display_frame = layer->GetDisplayFrame();
```
Comment (kalyankondapally) [inline common/display/displayplanemanager.cpp]: probably missed removing this ?

File: common/display/displayplanemanager.cpp [hunk 2]
```diff
@@ -175,7 +175,30 @@ std::tuple<bool, DisplayPlaneStateList> DisplayPlaneManager::ValidateLayers(
...
+        if (cursor_plane) {
+          commit_planes.emplace_back(OverlayPlane(cursor_plane, cursor_layer));
+          // Lets ensure we fall back to GPU composition in case
+          // cursor layer cannot be scanned out directly.
+          if (FallbacktoGPU(cursor_plane, cursor_layer, commit_planes)) {
+            cursor_plane = NULL;
+          }
+        }
+
+        // We need to do this here to avoid compositing cursor with any previous
```
Comment (kalyankondapally) [inline common/display/displayplanemanager.cpp]: u  can remove these comments now. Can u test on SKL too before commiting the changes

Comment (pallavigopal) [inline common/display/displayplanemanager.cpp]: ok tested on SKL things works fine

File: common/display/displayplanemanager.cpp (hunk 1 above)
Comment (pallavigopal) [inline common/display/displayplanemanager.cpp]: yes

**Pull request merged**